twarc_hashtags.py

```diff
@@ -62,20 +59,43 @@ def load(infile, outfile, db):
 
             data = json.loads(line)
             for tweet in ensure_flattened(data):
+                # Process Retweets:
+                if "referenced_tweets" in tweet:
+                    rts = [
+                        t
+                        for t in tweet["referenced_tweets"]
+                        if t["type"] == "retweeted"
+                    ]
+                    retweeted_tweet = rts[-1] if rts else None
+                    # If it's a native retweet, replace the "RT @user Text" with the original text, metrics, and entities, but keep the Author.
+                    if retweeted_tweet:
+                        # A retweet inherits everything from retweeted tweet.
+                        tweet["text"] = retweeted_tweet.pop(
+                            "text", tweet.pop("text", None)
+                        )
+                        tweet["entities"] = retweeted_tweet.pop(
```

Comment:
Since we are only interested in hashtags isn't entities all that is needed?

Reply:
Yes, but i'm inclined to leave them in because it's more explicit about what's happening - and in case there's any future additions / modifications that won't throw up any surprises. I don't think it hurts to have the extra bits there, but i'm also ok with just commenting them out.